### ISSUE TITLE
How to batch infer character image from a txt file?

### ISSUE THREAD

**Issue by ApolloZhangCN** [ecnuycxie/DG-Font#24]
Thanks for your wonderful work! I have finished the trainning, but I do not know how to infer  lots of character via a txt file by a special font type ?  I did these works:
1\   trained the repo and finished it.  python main.py --gpu 0 --img_size 80 --data_path ./data  --output_k 4 --batch_size 8 --val_num190
2\   use font2img.py to change the character into image , the dir is data, and the number  of character  is  52.

3\    python main.py --gpu 0 --img_size 80 --data_path ./data --output_k 4 --batch_size 8 --validation --load_model  GAN_20211130-124006.

4\  finnally get 32 images with only 10 character in them.  but  not  52 .   

question: 1\  how to get all of these character in  images.  2\ actually I just want the special style transfer,  example  font2 to  font 4 , how can I do this work? 

Could you help to give some advice ? Many thanks  for your kindly help!

**Comment by ecnuycxie** [comment]
source_id = 7
img_idx = 0
path_save = 'output'
ref_num = 10
        for ref_idx in range(len(evaluation_list)):
            for cnt_idx in range((int)(val_num/args.val_batch)):
                x_src = x_each_cls[source_id][args.val_batch*cnt_idx:(cnt_idx+1)*args.val_batch, :, :, :].cuda(args.gpu, non_blocking=True)
                rnd_idx = torch.randperm(x_each_cls[ref_idx].size(0))[:ref_num]
                x_ref_rnd = x_each_cls[ref_idx][rnd_idx].cuda(args.gpu, non_blocking=True)
                for sample_idx in range(args.val_batch):
                    x_ref_tmp = x_ref_rnd[sample_idx: sample_idx + ref_num].repeat((args.val_batch, 1, 1, 1))
                    c_src, skip1, skip2 = G_EMA.cnt_encoder(x_src)
                    s_ref = C_EMA(x_ref_tmp, sty=True)
                    s_ref = torch.mean(s_ref,dim=0)
                    s_ref = s_ref.unsqueeze(0)
                    x_res_ema_tmp,_ = G_EMA.decode(c_src, s_ref, skip1, skip2)
                    vutils.save_image(x_res_ema_tmp, os.path.join(path_save, '%d.png'%img_idx), normalize=True,
                                        nrow=1)
                    img_idx = img_idx+1
            img_idx = 0

**Comment by ApolloZhangCN** [comment]
&nbsp; &nbsp; thanks for your works!


&nbsp; &nbsp; I copy these codes into validation.py ,&nbsp; and change the&nbsp; "source_id = 7"&nbsp; to "source_id = 2" because of only 4 font in my project.&nbsp; &nbsp;create a dir named "output".&nbsp;
then run then main.py.&nbsp; &nbsp;thing goes good , but nothing output.&nbsp;


&nbsp; &nbsp; &nbsp;Hope any other adivce. thanks a lot!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "ecnuycxie/DG-Font"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月8日(星期三) 晚上7:23
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [ecnuycxie/DG-Font] How to batch infer character image from a txt file? (Issue #24)





 
source_id = 7
 img_idx = 0
 path_save = 'output'
 ref_num = 10
 for ref_idx in range(len(evaluation_list)):
 for cnt_idx in range((int)(val_num/args.val_batch)):
 x_src = x_each_cls[source_id][args.val_batch*cnt_idx:(cnt_idx+1)*args.val_batch, :, :, :].cuda(args.gpu, non_blocking=True)
 rnd_idx = torch.randperm(x_each_cls[ref_idx].size(0))[:ref_num]
 x_ref_rnd = x_each_cls[ref_idx][rnd_idx].cuda(args.gpu, non_blocking=True)
 for sample_idx in range(args.val_batch):
 x_ref_tmp = x_ref_rnd[sample_idx: sample_idx + ref_num].repeat((args.val_batch, 1, 1, 1))
 c_src, skip1, skip2 = G_EMA.cnt_encoder(x_src)
 s_ref = C_EMA(x_ref_tmp, sty=True)
 s_ref = torch.mean(s_ref,dim=0)
 s_ref = s_ref.unsqueeze(0)
 x_res_ema_tmp,_ = G_EMA.decode(c_src, s_ref, skip1, skip2)
 vutils.save_image(x_res_ema_tmp, os.path.join(path_save, '%d.png'%img_idx), normalize=True,
 nrow=1)
 img_idx = img_idx+1
 img_idx = 0
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
